include/deal.II/base/exceptions.h

```diff
@@ -1987,6 +2001,18 @@ namespace deal_II_exceptions
                                                                        dim2), \
          dealii::ExcDimensionMismatch((dim1), (dim2)))
 
+/**
+ * Special assertion for integer conversions.
+ *
+ * deal.II does not always use the same integer types as its dependencies. For
+ * example, PETSc uses signed integers whereas deal.II uses unsigned integers.
+ * This assertion checks that we can successfully convert between two index
+ * types.
+ */
+#define AssertIntegerConversion(index1, index2)                         \
```

Comment:
Realistically, these checks are only going to fail when running in release mode as you can not run large computations in a reasonable amount of time in debug mode. So, should these be AssertThrow()? Or do we add AssertThrow in one central location when we construct matrices/vectors/etc.?

Reply:
Not the individual index operations. This would slow down operations significantly. But when setting up data structures in a `reinit()` then we could `AssertThrow`.

Reply:
Ok.

Reply:
Fixed!